controller/src/main/java/org/jboss/as/controller/AttributeDefinition.java

```diff
@@ -134,6 +134,9 @@ private AttributeDefinition(String name, String xmlName, final ModelNode default
         if (defaultValue != null && defaultValue.isDefined()) {
             this.defaultValue = defaultValue;
             this.defaultValue.protect();
+            if (this.required) {
+                throw new IllegalArgumentException(); // WFCORE-2252. Incorrect combination from the code author
```

Comment:
This error should be updated to include a message, without a message a developer will always be required to identify the line number the exception was thrown from and check the code to identify the cause.